File: Two Pointers/LET_no1750_Minimum Length of String After Deleting Similar Ends.py
class Solution:
    def minimumLength(self, s: str) -> int:
        i, j = 0, len(s) - 1
        while i < j and s[i] == s[j]:
            cur = s[i]
            while i <= j and s[i] == cur:
                i += 1
            while i <= j and s[j] == cur:
                j -= 1
        
        return j - i + 1 if i <= j else 0

File: Two Pointers/Medium/LET_no1750_Minimum Length of String After Deleting Similar Ends.cpp
class Solution {
public:
    int minimumLength(string s) {
        int i = 0, j = s.size() - 1;
        while (i < j && s[i] == s[j]) {
            char cur = s[i];
            while (i <= j && s[i] == cur)
                i++;
            while (i <= j && s[j] == cur)
                j--;
        }

        return i <= j ? j - i + 1 : 0;
    }
};